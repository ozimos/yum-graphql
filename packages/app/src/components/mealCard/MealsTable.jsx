import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactTable from 'react-table';


class MealsTable extends Component {

  render() {

    const columns = [
      {
        Header: 'Meal Title',
        accessor: 'title',
        width: 275
      }, {
        Header: 'Quantity',
        accessor: 'MealOrders.quantity',
        width: 75
      }, {
        Header: () => (<span>Price (&#8358;)</span>),
        accessor: 'price',
        width: 175
      }, {
        Header: () => (<span>Subtotal (&#8358;)</span>),
        aggregate: vals => vals.reduce((accum, val) => accum + val),
        accessor: 'subTotal',
        width: 175
      }
    ];

    return (
      <div className="order-meal-modal" id="order-meal-modal">

        <div className="flexbox top">

          <h5 id="order-details">Order Details</h5>

          <button
            id="order-details-close"
            className="btn title-button"
            onClick={this.props.closeMealDetailModal}
          >
                &#10006;
          </button>

        </div>

        <div className="flexbox">

          <span>
        Total Charge:
          </span>

          <span>
            &#8358;{this.props.total}
          </span>

        </div>

        <ReactTable
          ref={(refReactTable) => { this.refReactTable = refReactTable; }}
          data={this.props.meals}
          columns={columns}
          minRows={0}
          defaultPageSize={this.props.mealsPagination.limit}
          loading={this.props.loading}
          pages={this.props.mealsPagination.pages}
          onFetchData={this.props.onFetchData}
          sortable={false}
          manual
          resizable={false}
          collapseOnSortingChange={false}
          collapseOnDataChange={false}
          collapseOnPageChange={false}
          nextText=">>"
          previousText="<<"
        />

      </div>
    );
  }
}

MealsTable.defaultProps = {
  mealsPagination: {
    pages: 1,
    limit: 5,
    offset: 0
  },
  loading: false,
  total: 0,
};

MealsTable.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  mealsPagination: PropTypes.shape({
    pages: PropTypes.number,
    limit: PropTypes.number,
    offset: PropTypes.number,
  }),
  onFetchData: PropTypes.func.isRequired,
  closeMealDetailModal: PropTypes.func.isRequired,
  loading: PropTypes.bool,
  total: PropTypes.number
};

export default MealsTable;
