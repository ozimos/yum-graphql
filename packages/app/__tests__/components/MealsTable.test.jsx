/* global React:false, shallow:false toJson:false */

import MealsTable from '../../src/components/mealCard/MealsTable';
import { rows } from '../__mocks__/mealDataMock';

const props = {
  mealsPagination: {
    limit: 10,
    offset: 0,
    pages: 1
  },
  currentOrderId: 'abc',
  loading: false,
  meals: rows,
  total: 0,
  closeMealDetailModal: jest.fn(),
  onFetchData: jest.fn(),
};

describe('MealsTable Component', () => {

  const setup = () => shallow(<MealsTable {...props} />);


  it('should render without throwing an error', () => {
    const wrapper = setup();
    expect(wrapper
      .exists(<table className="table" />)).toBe(true);
  });

  it('renders correctly', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

