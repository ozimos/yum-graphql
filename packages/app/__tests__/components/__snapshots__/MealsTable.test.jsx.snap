// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealsTable Component renders correctly 1`] = `
<div
  className="order-meal-modal"
  id="order-meal-modal"
>
  <div
    className="flexbox top"
  >
    <h5
      id="order-details"
    >
      Order Details
    </h5>
    <button
      className="btn title-button"
      id="order-details-close"
      onClick={[MockFunction]}
    >
      ✖
    </button>
  </div>
  <div
    className="flexbox"
  >
    <span>
      Total Charge:
    </span>
    <span>
      ₦
      0
    </span>
  </div>
  <ReactTable
    AggregatedComponent={[Function]}
    ExpanderComponent={[Function]}
    FilterComponent={[Function]}
    LoadingComponent={[Function]}
    NoDataComponent={[Function]}
    PadRowComponent={[Function]}
    PaginationComponent={[Function]}
    PivotValueComponent={[Function]}
    ResizerComponent={[Function]}
    TableComponent={[Function]}
    TbodyComponent={[Function]}
    TdComponent={[Function]}
    TfootComponent={[Function]}
    ThComponent={[Function]}
    TheadComponent={[Function]}
    TrComponent={[Function]}
    TrGroupComponent={[Function]}
    aggregatedKey="_aggregated"
    className=""
    collapseOnDataChange={false}
    collapseOnPageChange={false}
    collapseOnSortingChange={false}
    column={
      Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      }
    }
    columns={
      Array [
        Object {
          "Header": "Meal Title",
          "accessor": "title",
          "width": 275,
        },
        Object {
          "Header": "Quantity",
          "accessor": "MealOrders.quantity",
          "width": 75,
        },
        Object {
          "Header": [Function],
          "accessor": "price",
          "width": 175,
        },
        Object {
          "Header": [Function],
          "accessor": "subTotal",
          "aggregate": [Function],
          "width": 175,
        },
      ]
    }
    data={Array []}
    defaultExpanded={Object {}}
    defaultFilterMethod={[Function]}
    defaultFiltered={Array []}
    defaultPageSize={10}
    defaultResized={Array []}
    defaultSortDesc={false}
    defaultSortMethod={[Function]}
    defaultSorted={Array []}
    expanderDefaults={
      Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      }
    }
    filterable={false}
    freezeWhenExpanded={false}
    getLoadingProps={[Function]}
    getNoDataProps={[Function]}
    getPaginationProps={[Function]}
    getProps={[Function]}
    getResizerProps={[Function]}
    getTableProps={[Function]}
    getTbodyProps={[Function]}
    getTdProps={[Function]}
    getTfootProps={[Function]}
    getTfootTdProps={[Function]}
    getTfootTrProps={[Function]}
    getTheadFilterProps={[Function]}
    getTheadFilterThProps={[Function]}
    getTheadFilterTrProps={[Function]}
    getTheadGroupProps={[Function]}
    getTheadGroupThProps={[Function]}
    getTheadGroupTrProps={[Function]}
    getTheadProps={[Function]}
    getTheadThProps={[Function]}
    getTheadTrProps={[Function]}
    getTrGroupProps={[Function]}
    getTrProps={[Function]}
    groupedByPivotKey="_groupedByPivot"
    indexKey="_index"
    loading={false}
    loadingText="Loading..."
    manual={true}
    minRows={0}
    multiSort={true}
    nestingLevelKey="_nestingLevel"
    nextText=">>"
    noDataText="No rows found"
    ofText="of"
    onFetchData={[MockFunction]}
    originalKey="_original"
    pageSizeOptions={
      Array [
        5,
        10,
        20,
        25,
        50,
        100,
      ]
    }
    pageText="Page"
    pages={1}
    pivotDefaults={Object {}}
    pivotIDKey="_pivotID"
    pivotValKey="_pivotVal"
    previousText="<<"
    resizable={false}
    resolveData={[Function]}
    rowsText="rows"
    showPageJump={true}
    showPageSizeOptions={true}
    showPagination={true}
    showPaginationBottom={true}
    showPaginationTop={false}
    sortable={false}
    style={Object {}}
    subRowsKey="_subRows"
  />
</div>
`;
